src/quipucordsctl/commands/check.py

```diff
@@ -0,0 +1,297 @@
+    MISSING = "missing"
+
+
+@dataclass
+class PathCheckResult:
+    """Result of checking a path's status."""
+
+    status: StatusType
+    path: pathlib.Path
+    stat_info: Optional[os.stat_result] = None
+
+
+def get_help() -> str:
+    """Get the help/docstring for this command."""
+    return _(
+        "Check that all necessary files and directories exist for running "
+        "%(server_software_name)s."
+    ) % {"server_software_name": settings.SERVER_SOFTWARE_NAME}
+
+
+def check_directory_status(path: pathlib.Path) -> PathCheckResult:
+    """
+    Check the status of a directory.
+
+    Returns:
+        PathCheckResult with status indicating OK, bad permissions, wrong owner, or
+        missing.
+    """
+    try:
+        if not path.exists():
+            return PathCheckResult(StatusType.MISSING, path)
+
+        if not path.is_dir():
+            return PathCheckResult(StatusType.MISSING, path)
+
+        path_stat = path.stat()
+
+        # Check ownership - should be owned by current user
+        if path_stat.st_uid != os.getuid():
+            return PathCheckResult(StatusType.WRONG_OWNER, path, path_stat)
+
+        # Check permissions - should be readable and writable by owner
+        mode = path_stat.st_mode
+        if not (mode & stat.S_IRUSR and mode & stat.S_IWUSR):
+            return PathCheckResult(StatusType.BAD_PERMISSIONS, path, path_stat)
+
+        return PathCheckResult(StatusType.OK, path, path_stat)
+
+    except PermissionError:
+        return PathCheckResult(StatusType.BAD_PERMISSIONS, path)
+
+
+def check_file_status(path: pathlib.Path) -> PathCheckResult:
+    """
+    Check the status of a file.
+
+    Returns:
+        PathCheckResult with status indicating OK, bad permissions, wrong owner, or
+        missing.
+    """
+    try:
+        if not path.exists():
+            return PathCheckResult(StatusType.MISSING, path)
+
+        if not path.is_file():
+            return PathCheckResult(StatusType.MISSING, path)
+
+        path_stat = path.stat()
+
+        # Check ownership - should be owned by current user
+        if path_stat.st_uid != os.getuid():
+            return PathCheckResult(StatusType.WRONG_OWNER, path, path_stat)
+
+        # Check permissions - should be readable and writable by owner
+        mode = path_stat.st_mode
+        if not (mode & stat.S_IRUSR and mode & stat.S_IWUSR):
+            return PathCheckResult(StatusType.BAD_PERMISSIONS, path, path_stat)
+
+        return PathCheckResult(StatusType.OK, path, path_stat)
+
+    except PermissionError:
+        return PathCheckResult(StatusType.BAD_PERMISSIONS, path)
+
+
+def _log_ok_status(result: PathCheckResult) -> None:
+    """Log OK status with detailed info if available."""
+    try:
+        if result.stat_info:
+            logger.info(
+                "%(path)s: OK: %(perms)s %(owner)s",
+                {
+                    "path": result.path,
+                    "perms": oct(result.stat_info.st_mode)[-3:],
```

Comment:
Unfortunate that Python doesn't seem to provide better way to turn that into more familiar representation.

How do you feel about this?
```suggestion
import stat  # that would need to go to top
                    "perms": stat.filemode(result.stat_info.st_mode),
```

It will print `-rw-r--r--` instead of `644`, which is different than your version. But the benefit is that someone else wrote that code and if there are any edge cases we haven't thought about, there's a good chance they are already covered.